diffrax/_integrate.py

```diff
@@ -136,18 +137,22 @@ def _check(term_cls, term, term_contr_kwargs, yi):
                 for term, arg, term_contr_kwarg in zip(
                     term.terms, get_args(_tmp), term_contr_kwargs
                 ):
-                    if not _term_compatible(yi, args, term, arg, term_contr_kwarg):
+                    if not _term_compatible(
+                        yi, args, term, arg, term_contr_kwarg, bwd_compat
+                    ):
```

Comment:
I think this is now wrong, as the return type is a 2-tuple rather than a boolean?

I think perhaps we should instead change the name and signature of this function to be `_assert_term_compatible(...) -> None`, and have it raise an error if it finds an issue?
In a couple of cases (here; in the backward compatibility), we can then try-except the error in question. (Here to do a `raise ValueError(...) from e`; in the backward compability just to treat it as a boolean.)

Reply:
Sure, changing to assert makes sense, then I can make the try catch block explicitly around the backward compatibility code rather than as a flag. It's not hurdle keeping it, but I'm curious what's the motivation for this specific backward compatible section? Diffrax (and equinox) have introduced breaking/non backward compatible changes (I think new events did for example, and equinox is about with jax, and I'm sure other cases).

Reply:
Really just that I try to avoid backward compatibility breaks if it's reasonable to do so.

The change with events, for example, includes a similar backward compatibility converter. That one was a little imperfect because we no longer provide the same information as the old events API (which was far too broad), but was still a best-effort!

Reply:
I admire your commitment to backwards compatibility for pre 1.0 releases, in the crucible I was forged (qiskit) there were so many breaking changes in minor releases pre 1.0 that they literally made a YouTube series called breaking changes to help explain them all every minor release.